src/documentation.js

```diff
@@ -26,13 +26,70 @@ function _mapToObj(map) {
   return returnObj;
 }
 
+/**
+ * maps requests to delete to an array containing the promise and the body of the request
+ * @param {aws} provider
+ * @param {documentationPieces} Object, contains documentationToUpsert and documentationToDelete
+ * @returns request type array
+ */
+function createDocumentationRequests(aws, { documentationToUpsert, documentationToDelete }) {
+  let requests = []
+  // delete documentation parts
+  requests = requests.concat(
+    documentationToDelete.map(part => {
+      return{ 
+        request: aws.request('APIGateway', 'deleteDocumentationPart', {
+          documentationPartId: part.id,
+          restApiId: part.restApiId,
+        }),
+        part
+      }
+    })
+  )
+  // upsert documentation parts
+  requests = requests.concat(
+    documentationToUpsert.map(part => {
+      return {
+        request: aws.request('APIGateway', 'createDocumentationPart', {
```

Comment:
I believe this is where we are getting rate limited. anyway to catch those and slow down?